### DISCUSSION TITLE
Release v2.0.0rc5

### DISCUSSION THREAD

**Discussion by miurahr** [miurahr/aqtinstall#404] [Announcements]
<hr /><em>This discussion was created from the release <a href='https://github.com/miurahr/aqtinstall/releases/tag/v2.0.0rc5'>Reelase v2.0.0rc5</a>.</em>

**Comment by miurahr** [comment]
I'd like to set a target date of general release v2.0.0 in 29, Sep. 2021.

A day 29  is known as "The meat 🥩 day" in Japan, taking a Japanese pronounce ni-ku, and many steak/BBQ/beef bowl restaurants are in sale. Let's prepare a party! 🍖

**Reply by pzhlkj6612** [reply]
Go for it! More users will make developers discover more problems, whether in the code or documentation.

**Comment by ddalcino** [comment]
I don't want to ruin the party, but it looks like the [upload for the continuous build](https://github.com/miurahr/aqtinstall/runs/3711190741?check_suite_focus=true#step:7:29) failed:

```
Uploading release artifacts
  Error: .gravitybee\dist\latest\aqt.exe doesn't match any files
```

This is the sequence of events, as far as I can tell:
1. `ci/build_standalone.bat` runs, builds an executable, and renames it `.gravitybee\dist\latest\aqt.exe`
2. The `svenstaro/upload-release-action@v2` action runs, and uploads `.gravitybee\dist\latest\aqt.exe` to the release for 2.0.0rc5
3. ???
4. The `marvinpinto/action-automatic-releases@latest` action runs, but cannot locate `.gravitybee\dist\latest\aqt.exe`.

I wonder if the `marvinpinto/action-automatic-releases@latest` action is somehow misinterpreting the Windows directory separators as escape sequences? :confused: I don't understand how the file could have disappeared between the two actions.

**Reply by pzhlkj6612** [reply]
I'm going to adjust the CI scripts so that they fail when no files are uploaded.

**Reply by miurahr** [reply]
The actions doing https://github.com/marvinpinto/actions/blob/master/packages/automatic-releases/src/uploadReleaseArtifacts.ts
```
 for (const fileGlob of files) {
    const paths = await globby(fileGlob);
```
So files parameter is not a list of files but a list of glob patterns. IMHO, it may be able to be `**/aqt.exe`

**Reply by pzhlkj6612** [reply]
Hey, use https://github.com/marvinpinto/actions/blob/05a5fd79e119b74bf8fa909a7ecb52bef9aaf1da/packages/automatic-releases/src/uploadReleaseArtifacts.ts#L14-L15 instead, please.

**Reply by ddalcino** [reply]
Would it be possible to just use the `svenstaro/upload-release-action` twice? It looks like it allows you to specify a release tag and overwrite an existing file. Could that work for the continuous release?

**Reply by pzhlkj6612** [reply]
> So files parameter is not a list of files but a list of glob patterns. IMHO, it may be able to be `**/aqt.exe`

But when no file is found, it still won't fail the workflow, right?

I think I've figured it out. "marvinpinto/action-automatic-releases" should throw an error instead of using [`core.error()`](https://github.com/actions/toolkit/blob/ea81280a4d48fb0308d40f8f12ae00d117f8acb9/packages/core/README.md#logging) (it just do logging) at [uploadReleaseArtifacts.ts#L17](https://github.com/marvinpinto/actions/blob/05a5fd79e119b74bf8fa909a7ecb52bef9aaf1da/packages/automatic-releases/src/uploadReleaseArtifacts.ts#L17) , otherwise the error handling code at [main.ts#L322](https://github.com/marvinpinto/actions/blob/05a5fd79e119b74bf8fa909a7ecb52bef9aaf1da/packages/automatic-releases/src/main.ts#L322) will not work. However, I don't know if it's the author's intention to report success when nothing was found.

See also marvinpinto/actions#62 and marvinpinto/actions#63 .

**Reply by pzhlkj6612** [reply]
> Would it be possible to just use the `svenstaro/upload-release-action` twice? It looks like it allows you to specify a release tag and overwrite an existing file. Could that work for the continuous release?

I agree to use "[svenstaro/upload-release-action](https://github.com/svenstaro/upload-release-action)". It will stop the workflow when nothing is found ([main.ts#L166](https://github.com/svenstaro/upload-release-action/blob/a70d06e688338cf4b2580a56cc04019b6f44f545/src/main.ts#L166)).

**Reply by pzhlkj6612** [reply]
Hold on a second.

> It will stop the workflow when nothing is found ([main.ts#L166](https://github.com/svenstaro/upload-release-action/blob/a70d06e688338cf4b2580a56cc04019b6f44f545/src/main.ts#L166)).

The code I referenced takes effect when `file_glob` is enabled. If `file_glob` is disabled (by default) and the `file` parameter points to a non-existent file, it will also fail the workflow ( [`fs.statSync(file)`](https://nodejs.org/api/fs.html#fs_fs_statsync_path_options) at [main.ts#L54](https://github.com/svenstaro/upload-release-action/blob/a70d06e688338cf4b2580a56cc04019b6f44f545/src/main.ts#L54)).

**Reply by pzhlkj6612** [reply]
#405

**Reply by miurahr** [reply]
merged and manually uploaded aqt.exe to releases

**Reply by pzhlkj6612** [reply]
About `marvinpinto/action-automatic-releases@latest` . I've made marvinpinto/actions#387 and marvinpinto/actions#388 . If one of them get merged, we could go back to use `marvinpinto/action-automatic-releases` .

**Comment by pzhlkj6612** [comment]
Shall we append a version number to the file name of the standalone exe in the normal release (not continuous)? For example, upload a file named "aqt-v2.0.0rc5.exe" in release v2.0.0rc5.

**Reply by pzhlkj6612** [reply]
One thing I forgot to mention is that we should use "aqt.exe" as the filename in continuous releases.

Any opinions?

**Reply by pzhlkj6612** [reply]
OK, I think this could be done in https://github.com/miurahr/aqtinstall/issues/407#issuecomment-928664401 . Version 2.0.0 should not be blocked by this.

**Comment by miurahr** [comment]
Pushes v2 release commit now.

**Reply by pzhlkj6612** [reply]
And please pay attention to the spelling of the word "release":

```diff
- Reelase
+ Release
```

- https://github.com/miurahr/aqtinstall/releases/tag/v2.0.0rc3
- https://github.com/miurahr/aqtinstall/releases/tag/v2.0.0rc5

**Reply by miurahr** [reply]
Failure in test.pypi upload is not a problem.
